### PR TITLE
fix: mode variant add null and inherit from parent to components

### PR DESCRIPTION
**Pull request by EmelieLitwin**
**Describe pull-request**  
Card, message, textfield and textarea was missing null option to mode variant and/or Inherit from parent as option in storybook

**How to test**  
Check the components and that mode variants working in storybook and looks ok in code

### CODE REVIEW

**Comment by charlottelorensson** [comment]
Skärmavbild 2023-02-09 kl  16 48 09

Perhaps not the place (and perhaps won't be so visible) but just something I noticed, it looks like the web component has blue lines on the corners.

And it seems like the font color in the textfields are not aligned with Figma:

Skärmavbild 2023-02-09 kl  17 00 10


But the mode variants seem to be working fine! :)

**Comment by charlottelorensson** [comment]
Forgot to mention that the first image was the message component* :)

File: tegel/src/components/textarea/textarea.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ export class Textarea {
   @Prop() maxLength: number;
 
   /** Mode variant of the textarea */
-  @Prop() modeVariant: 'primary' | 'secondary' = 'primary';
+  @Prop() modeVariant: 'primary' | 'secondary' | null = null;
```
Comment (mJarsater) [inline tegel/src/components/textarea/textarea.tsx:49]:
```suggestion
  @Prop() modeVariant: 'primary' | 'secondary' = null;
```

File: tegel/src/components/message/sdds-message.tsx [hunk 2]
```diff
@@ -10,7 +10,7 @@ export class SddsMessage {
   @Prop() header: string;
 
   /** Variant of the component, based on current mode. */
-  @Prop() modeVariant: 'primary' | 'secondary' = 'primary';
+  @Prop() modeVariant: 'primary' | 'secondary' | null = null;
```
Comment (mJarsater) [inline tegel/src/components/message/sdds-message.tsx:13]:
```suggestion
  @Prop() modeVariant: 'primary' | 'secondary' = null;
```

**Review by mJarsater — CHANGES_REQUESTED** [review]
Nice 🚀! I think we can remove the optional null value on the modeVariant prop. Then it won't show in the readme and Im guessing we don't really want people to have to care about this.

**Comment by EmelieLitwin** [comment]
> Nice 🚀! I think we can remove the optional null value on the modeVariant prop. Then it won't show in the readme and Im guessing we don't really want people to have to care about this.

Good feedback! If that's something you think we should do, it should be done on all components!

**Comment by mJarsater** [comment]
> > Nice 🚀! I think we can remove the optional null value on the modeVariant prop. Then it won't show in the readme and Im guessing we don't really want people to have to care about this.
> 
> Good feedback! If that's something you think we should do, it should be done on all components!

Yeah that makes sense to me. Then all components with a modeVariant would have:
`@Prop() modeVariant: 'primary' | 'secondary' = null;`

**Comment by EmelieLitwin** [comment]
> Skärmavbild 2023-02-09 kl 16 48 09
> 
> Perhaps not the place (and perhaps won't be so visible) but just something I noticed, it looks like the web component has blue lines on the corners.
> 
> And it seems like the font color in the textfields are not aligned with Figma:
> 
> Skärmavbild 2023-02-09 kl 17 00 10
> 
> But the mode variants seem to be working fine! :)

What is he problem with the textfield? In dark mode the inside label is grey-400 and placeholder grey-600, looks like it's the same in figma?

**Review by mJarsater — APPROVED** [review]
LGTM!

**Pull request merged**